common/src/java/org/apache/hadoop/hive/conf/HiveConfUtil.java

```diff
@@ -195,20 +196,31 @@ public static void updateJobCredentialProviders(Configuration jobConf) {
           + " previous location was " + oldKeyStoreLocation);
     }
 
+    updateCredentialProviderPasswordForJobs(jobConf);
+  }
+
+  public static void updateCredentialProviderPasswordForJobs(Configuration jobConf) {
     String credstorePassword = getJobCredentialProviderPassword(jobConf);
     if (credstorePassword != null) {
       String execEngine = jobConf.get(ConfVars.HIVE_EXECUTION_ENGINE.varname);
 
-      if ("mr".equalsIgnoreCase(execEngine)) {
-        // if the execution engine is MR set the map/reduce env with the credential store password
-
+      if ("mr".equalsIgnoreCase(execEngine) || "tez".equalsIgnoreCase(execEngine)) {
```

Comment:
do we have enums for the execution engines?

Reply:
I don't think so :) I think we should introduce it someday and remove the usage of this string madness from at least hive-exec